pkg/graveler/settings/manager.go

```diff
@@ -134,30 +138,22 @@ func (m *Manager) getFromStore(ctx context.Context, repositoryID graveler.Reposi
 }
 
 // UpdateWithLock atomically gets a setting, performs the update function, and persists the setting to the store.
-// The emptySetting parameter is used to determine the type passed to the update function.
-func (m *Manager) UpdateWithLock(ctx context.Context, repositoryID graveler.RepositoryID, key string, emptySetting proto.Message, update func(setting proto.Message)) error {
+// The settingTemplate parameter is used to determine the type passed to the update function.
+func (m *Manager) UpdateWithLock(ctx context.Context, repositoryID graveler.RepositoryID, key string, settingTemplate proto.Message, update func(proto.Message)) error {
 	repo, err := m.refManager.GetRepository(ctx, repositoryID)
 	if err != nil {
 		return err
 	}
 	_, err = m.branchLock.MetadataUpdater(ctx, repositoryID, repo.DefaultBranchID, func() (interface{}, error) {
```

Comment:
Just wanted to say that at this point it looked strange at first: we use the branch longer to make sure we are the only one that updates - but do not use it to make sure nobody tries to read it while we write.
But I assume because we depend on the same object and we write just one - there will be no issue to just read the latest because the object store will make sure it will be there and valid.